back/app/services/participation_context_service.rb

```diff
@@ -306,7 +306,14 @@ def mode_specific_idea_voting_disabled_reason(mode, context, user)
   end
 
   def posting_limit_reached?(context, user)
-    context.posting_limited? && context.ideas.where(author: user).size >= context.posting_limited_max
+    return true if context.posting_limited? && context.ideas.where(author: user).size >= context.posting_limited_max
+
+    if context.native_survey? && context.allow_anonymous_participation?
```

Comment:
We use it elsewhere in the same file - Line 88:
`elsif !context.ideation? && !context.native_survey?
      POSTING_DISABLED_REASONS[:not_ideation] # TODO: (native surveys) change reason code?`
      
And in this case I wanted to avoid generating a hash unless it was native survey. I guess `posting_limited?` will at least stop it being generated in most cases, I'll change